src/core/geometry/qgscircularstring.h

```diff
@@ -50,6 +50,23 @@ class CORE_EXPORT QgsCircularString: public QgsCurve
                        const QgsPoint &p2,
                        const QgsPoint &p3 ) SIP_HOLDGIL;
 
+    /**
+     * Construct a circular string from arrays of coordinates. If the z or m
+     * arrays are non-empty then the resultant circular string will have
+     * z and m types accordingly.
```

Comment:
Can you also indicate that the number of points will be the minimum of points between x and y?

Reply:
Done!